### DISCUSSION TITLE
Creating "Science"-style figure captions

### DISCUSSION THREAD

**Discussion by january3** [quarto-dev/quarto-cli#2907] [Q&A]
I am attempting to mimick the "Science" style of figure captions. For this, the first sentence of the figure caption, including the caption title and figure number should be in bold. I can make the figure caption bold by defining `fig-title: "**Figure**"`; similarly, the first sentence of the caption can be made bold directly. However, the number (and the separator) are not shown in bold. Is there any way to achieve what I want in quarto?

I tried to define fig-title: '**Figure' to start the bold marking and then end the markdown bold after the first sentence of the caption, but that didn't work either – the stars are not interpreted as markdown and simply show in the HTML directly.

**Comment by jimjam-slam** [comment]
It's not quite the same thing, unfortunately, but if you wanted to target the first _line_ (not the first sentence) in an HTML-based output, you could add the following CSS block to your document:

```html
<style>
  figcaption.figure-caption::first-line {
    font-weight: bold;
  }
</style>
```

I think to target the first sentence, you're going to need to wrap it in a `<span>`. You could try to do that manually in your documents by writing the first sentence of every figure caption `[using the span syntax]{class="captionlead"}` (although I'm not sure how well spans variously work inside figure captions, table captions, code block captions etc.) and then styling the class `.captionlead` as above. For example:

```
![[Here's the first sentence.]{class="captionlead"} And another that explains stuff more.](/my/figure.png)
```

Otherwise, to automate it (and perhaps to make it work if Quarto can't handle adding spans in some places), you could [write a Lua filter](https://quarto.org/docs/extensions/filters.html) that targets figure captions, adds the start of a span, and walks along until it finds a full stop, then ends the span.

**Reply by january3** [reply]
It would be easier if "fig-title" could be specified with a placeholder with a number, for example 

    fig-title: "Figure #."
